build.gradle

```diff
@@ -15,6 +15,13 @@ allprojects {
             toolchain.languageVersion = JavaLanguageVersion.of(21)
         }
     }
+
+    // Required for Hilt to use the toolchain
+    tasks.withType(JavaCompile).configureEach {
+        javaCompiler = javaToolchains.compilerFor {
+            languageVersion = JavaLanguageVersion.of(21)
+        }
+    }
```

Comment:
```suggestion
```

Reply:
We don't need this config, it's honored from `JavaPluginExtension` config above, see also https://github.com/gradle/gradle/blob/95a2521ca265080e9f3c12ef0a25d10c6ff3da22/platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java#L229

Reply:
Plugins inheriting from `JavaBasePlugin` configure the toolchain properly, yes, but the Hilt plugin does not.
See https://github.com/google/dagger/blob/9a67471586c3ed1bccb5d0e13a86af06e024cd1a/java/dagger/hilt/android/plugin/main/src/main/kotlin/dagger/hilt/android/plugin/HiltGradlePlugin.kt#L367